### ISSUE TITLE
[Bug] Migration Error in Logs during startup

### ISSUE THREAD

**Issue by jeffcorpuz** [reactioncommerce/reaction#4897]
## Issue Description
- Running fresh installation steps via docker-compose or through reaction platform outputs a migration warning. 
- Reaction Version isn't stated correctly in INFO.

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. Run `docker-compose up -d` or `reaction-platform/make` on a fresh Git repo
2. Echo out logs of the `reaction` container to see warnings on migration 24, 32
3. Notice incorrect Reaction Version : 2.0.0-rc.7 
4. To verify `sh` into Docker container and run `git log` to double check it is properly reflecting rc.8 history

```
03:22:53.744Z  INFO Reaction: Reaction Version: 2.0.0-rc.7
03:22:53.958Z  INFO Reaction: Migrating from version 0 -> 50
03:22:53.958Z  INFO Reaction: Running up() on version 1
03:22:54.005Z  INFO Reaction: Running up() on version 2
03:22:54.020Z  INFO Reaction: Running up() on version 3
03:22:54.065Z  WARN Reaction: Skipped loading settings from reaction.json.
03:22:55.947Z  INFO Reaction: Running up() on version 4
03:22:55.951Z  INFO Reaction: Running up() on version 5
03:22:56.006Z  INFO Reaction: Running up() on version 6
03:22:56.009Z  INFO Reaction: Running up() on version 7
03:22:56.020Z  INFO Reaction: Running up() on version 8
03:22:56.106Z  INFO Reaction: Running up() on version 9
03:22:56.117Z  INFO Reaction: Running up() on version 10
03:22:56.121Z  INFO Reaction: Running up() on version 11
03:22:56.129Z  INFO Reaction: Running up() on version 12
03:22:56.130Z  INFO Reaction: Running up() on version 13
03:22:56.133Z  INFO Reaction: Running up() on version 14
03:22:56.137Z  INFO Reaction: Running up() on version 15
03:22:56.150Z  INFO Reaction: Running up() on version 16
03:22:56.152Z  INFO Reaction: Running up() on version 17
03:22:56.153Z  INFO Reaction: Running up() on version 18
03:22:56.228Z  INFO Reaction: Running up() on version 19
03:22:56.253Z  INFO Reaction: Running up() on version 20
03:22:56.264Z  INFO Reaction: Running up() on version 21
03:22:56.265Z  INFO Reaction: Running up() on version 22
03:22:56.267Z  INFO Reaction: Running up() on version 23
03:22:56.271Z  INFO Reaction: Running up() on version 24
03:22:56.355Z ERROR Reaction:
  Error in migration 24, publishProductsToCatalog TypeError: Cannot read property 'emit' of undefined
      at Promise.asyncApply (imports/plugins/core/catalog/server/no-meteor/utils/publishProductToCatalog.js:80:15)
      at /home/node/.meteor/packages/promise/.0.11.1.lt6vo5.rl7ft++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
   => awaited here:
      at Function.Promise.await (/home/node/.meteor/packages/promise/.0.11.1.lt6vo5.rl7ft++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
      at Promise.asyncApply (imports/plugins/core/catalog/server/no-meteor/utils/publishProductsToCatalog.js:13:16)
      at /home/node/.meteor/packages/promise/.0.11.1.lt6vo5.rl7ft++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
   => awaited here:
      at Function.Promise.await (/home/node/.meteor/packages/promise/.0.11.1.lt6vo5.rl7ft++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
      at Object.up (imports/plugins/core/versions/server/migrations/24_publish_all_existing_visible_products.js:23:17)
      at migrate (packages/percolate_migrations.js:233:25)
      at Object.Migrations._migrateTo (packages/percolate_migrations.js:253:7)
      at Object.Migrations.migrateTo (packages/percolate_migrations.js:167:10)
      at Hooks.Events.add (imports/plugins/core/versions/server/startup.js:31:18)
      at /opt/reaction/src/node_modules/@reactioncommerce/hooks/dist/main.js:71:14
      at Array.reduce (<anonymous>)
      at Object.Hooks.Events.run (/opt/reaction/src/node_modules/@reactioncommerce/hooks/dist/main.js:70:22)
      at Object.init (imports/plugins/core/core/server/Reaction/core.js:84:18)
      at startup (imports/plugins/core/core/server/startup/index.js:38:12)
      at Function.time (/opt/reaction/src/.meteor/local/build/programs/server/profile.js:309:28)
      at /opt/reaction/src/.meteor/local/build/programs/server/boot.js:427:13
      at /opt/reaction/src/.meteor/local/build/programs/server/boot.js:472:5
      at Function.run (/opt/reaction/src/.meteor/local/build/programs/server/profile.js:510:12)
      at /opt/reaction/src/.meteor/local/build/programs/server/boot.js:470:11
03:22:56.356Z ERROR Reaction: Migration 24 failed to create catalog products for some published products.
03:22:56.356Z  INFO Reaction: Running up() on version 25
03:22:56.366Z  INFO Reaction: Running up() on version 26
03:22:56.367Z  INFO Reaction: Running up() on version 27
03:22:56.368Z  INFO Reaction: Running up() on version 28
03:22:56.372Z  INFO Reaction: Running up() on version 29
03:22:56.378Z  INFO Reaction: Running up() on version 30
03:22:56.384Z  INFO Reaction: Running up() on version 31
03:22:56.387Z  INFO Reaction: Running up() on version 32
03:22:56.420Z ERROR Reaction:
  Error in migration 32, publishProductsToCatalog TypeError: Cannot read property 'emit' of undefined
      at Promise.asyncApply (imports/plugins/core/catalog/server/no-meteor/utils/publishProductToCatalog.js:80:15)
      at /home/node/.meteor/packages/promise/.0.11.1.lt6vo5.rl7ft++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
   => awaited here:
      at Function.Promise.await (/home/node/.meteor/packages/promise/.0.11.1.lt6vo5.rl7ft++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
      at Promise.asyncApply (imports/plugins/core/catalog/server/no-meteor/utils/publishProductsToCatalog.js:13:16)
      at /home/node/.meteor/packages/promise/.0.11.1.lt6vo5.rl7ft++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
   => awaited here:
      at Function.Promise.await (/home/node/.meteor/packages/promise/.0.11.1.lt6vo5.rl7ft++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
      at Object.up (imports/plugins/core/versions/server/migrations/32_publish_all_existing_visible_products.js:23:17)
      at migrate (packages/percolate_migrations.js:233:25)
      at Object.Migrations._migrateTo (packages/percolate_migrations.js:253:7)
      at Object.Migrations.migrateTo (packages/percolate_migrations.js:167:10)
      at Hooks.Events.add (imports/plugins/core/versions/server/startup.js:31:18)
      at /opt/reaction/src/node_modules/@reactioncommerce/hooks/dist/main.js:71:14
      at Array.reduce (<anonymous>)
      at Object.Hooks.Events.run (/opt/reaction/src/node_modules/@reactioncommerce/hooks/dist/main.js:70:22)
      at Object.init (imports/plugins/core/core/server/Reaction/core.js:84:18)
      at startup (imports/plugins/core/core/server/startup/index.js:38:12)
      at Function.time (/opt/reaction/src/.meteor/local/build/programs/server/profile.js:309:28)
      at /opt/reaction/src/.meteor/local/build/programs/server/boot.js:427:13
      at /opt/reaction/src/.meteor/local/build/programs/server/boot.js:472:5
      at Function.run (/opt/reaction/src/.meteor/local/build/programs/server/profile.js:510:12)
      at /opt/reaction/src/.meteor/local/build/programs/server/boot.js:470:11
03:22:56.420Z ERROR Reaction: Migration 32 failed to create catalog products for some published products.
03:22:56.420Z  INFO Reaction: Running up() on version 33
03:22:56.423Z  INFO Reaction: Running up() on version 34
03:22:56.424Z  INFO Reaction: Running up() on version 35
03:22:56.425Z  INFO Reaction: Running up() on version 36
03:22:56.427Z  INFO Reaction: Running up() on version 37
03:22:56.430Z  INFO Reaction: Running up() on version 38
03:22:56.431Z  INFO Reaction: Running up() on version 39
03:22:56.431Z  INFO Reaction: Running up() on version 40
03:22:56.443Z  INFO Reaction: Running up() on version 41
03:22:56.444Z  INFO Reaction: Running up() on version 42
03:22:56.465Z  INFO Reaction: Running up() on version 43
03:22:56.473Z  INFO Reaction: Running up() on version 43.1
03:22:56.478Z  INFO Reaction: Running up() on version 44
03:22:56.479Z  INFO Reaction: Running up() on version 45
03:22:56.489Z  INFO Reaction: Running up() on version 46
03:22:56.490Z  INFO Reaction: Running up() on version 47
03:22:56.491Z  INFO Reaction: Running up() on version 48
03:22:56.497Z  INFO Reaction: Running up() on version 49
03:22:56.498Z  INFO Reaction: Running up() on version 50
03:22:56.505Z  INFO Reaction: Finished migrating.
03:22:56.514Z  WARN Reaction: OpenExchangeRates API not configured. Not adding fetchRates job
03:22:56.521Z  WARN Reaction: OpenExchangeRates API not configured. Not adding flushRates job
03:22:58.271Z  INFO Reaction: Reaction initialization finished: 9617ms
03:22:58.470Z  INFO Reaction: GraphQL listening at http://localhost:3000/graphql-alpha
03:22:58.471Z  INFO Reaction: GraphiQL UI: http://localhost:3000/graphiql
=> Started your app.

=> App running at: http://localhost:3000
Note: you are using a pure-JavaScript implementation of bcrypt.
While this implementation will work correctly, it is known to be
approximately three times slower than the native implementation.
In order to use the native implementation instead, run
```

### Versions
2.0.0-rc.8

**Comment by zenweasel** [comment]
Why do you think the version number is incorrect? It typically doesn't change until it's actually released

**Comment by jeffcorpuz** [comment]
It was always matching the release prior to this

**Comment by spencern** [comment]
I do need to bump the version still, that's why you're seeing `rc.7` here. We should investigate the migration errors.

**Comment by aldeed** [comment]
I see the issue. I'll do a fix shortly

**Comment by spencern** [comment]
Closed by #4907
